src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -331,34 +327,30 @@ private static Uri CreateConnectUrl(Uri url, string connectionId)
 
         private async Task StartTransport(Uri connectUrl, HttpTransportType transportType, TransferFormat transferFormat)
         {
-            // Create the pipe pair (Application's writer is connected to Transport's reader, and vice versa)
-            var options = new PipeOptions(writerScheduler: PipeScheduler.ThreadPool, readerScheduler: PipeScheduler.ThreadPool, useSynchronizationContext: false, pauseWriterThreshold: 0, resumeWriterThreshold: 0);
-            var pair = DuplexPipe.CreateConnectionPair(options, options);
-
             // Construct the transport
             var transport = _transportFactory.CreateTransport(transportType);
 
             // Start the transport, giving it one end of the pipe
             try
             {
-                await transport.StartAsync(connectUrl, pair.Application, transferFormat, this);
+                await transport.StartAsync(connectUrl, transferFormat);
             }
             catch (Exception ex)
             {
                 Log.ErrorStartingTransport(_logger, transport, ex);
 
-                // Clean up pipes and null out transport when we fail to start.
-                pair.Transport.Input.Complete();
-                pair.Transport.Output.Complete();
-                pair.Application.Input.Complete();
-                pair.Application.Output.Complete();
                 _transport = null;
                 throw;
             }
 
+            if (transportType == HttpTransportType.LongPolling)
+            {
+                // Disable keep alives for long polling
+                Features.Set<IConnectionInherentKeepAliveFeature>(new ConnectionInherentKeepAliveFeature(_httpClient.Timeout));
```

Comment:
You had to move this here because a transport is no longer passed the connection, but it feels like the implementation detail of long polling is leaking into the generic HttpConnection class. Is there a better way?

Reply:
This mirrors the server side and I think it's fine to have it here. The transport doesn't care about features, the HttpConnection layer is coupled to knowledge of specific transports and is where the concept of features is introduced. This is IMO, the right place.